backend/src/main/java/com/redhat/cloud/notifications/routers/ReadyCheck.java

```diff
@@ -0,0 +1,86 @@
+package com.redhat.cloud.notifications.routers;
+
+import com.redhat.cloud.notifications.models.CamelProperties;
+import com.redhat.cloud.notifications.models.Endpoint;
+import com.redhat.cloud.notifications.models.EndpointStatus;
+import com.redhat.cloud.notifications.models.EndpointType;
+import com.redhat.cloud.notifications.openbridge.Bridge;
+import com.redhat.cloud.notifications.openbridge.BridgeApiService;
+import com.redhat.cloud.notifications.openbridge.BridgeAuth;
+import io.quarkus.scheduler.Scheduled;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.jboss.logging.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+import javax.transaction.Transactional;
+import javax.ws.rs.WebApplicationException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Scheduled task to check (OpenBridge) Endpoints
+ * for their state. Those endpoints can go from
+ * 'accepted' to 'provisioning' and then 'ready' or 'failed'.
+ * We need to poll OB regularly to see when they are
+ * good and then set the state accordingly.
+ */
+@ApplicationScoped
+public class ReadyCheck  {
+
+    private static final Logger LOGGER = Logger.getLogger(ReadyCheck.class);
+
+    @Inject
+    EntityManager em;
+
+    @Inject
+    @RestClient
+    BridgeApiService bridgeApiService;
+
+    @Inject
+    Bridge bridge;
+
+    @Inject
+    BridgeAuth bridgeAuth;
+
+    String endpointQueryString = "SELECT e FROM Endpoint e " +
+            "WHERE e.compositeType.type = :type AND e.compositeType.subType IN (:subTypes) " +
+            "AND e.status NOT IN (:ready, :failed) ";
+
+    @Scheduled(concurrentExecution = Scheduled.ConcurrentExecution.SKIP, every = "10s")
+    @Transactional
+    public void execute() {
+
+        Query query = em.createQuery(endpointQueryString);
+        query.setParameter("ready", EndpointStatus.READY);
+        query.setParameter("failed", EndpointStatus.FAILED);
+        query.setParameter("type", EndpointType.CAMEL);
+        query.setParameter("subTypes", "slack");
```

Comment:
We need Postgres' `SELECT FOR UPDATE` and `SKIP LOCKED` here to avoid concurrency issues when the query will be executed by multiple pods at the same time:
```suggestion
        query.setParameter("subTypes", "slack");
        query.setLockMode(LockModeType.PESSIMISTIC_WRITE);
        query.setHint("javax.persistence.lock.timeout", LockOptions.SKIP_LOCKED);
```
⚠️ My suggestion is untested, we need to make sure that Hibernate ORM adds `FOR UPDATE` and `SKIP LOCKED` in the generated SQL.

Reply:
To be checked in logs.

Reply:
The Hibernate query looks good:
```
select
    endpoint0_.id as id1_9_,
    endpoint0_.created as created2_9_,
    endpoint0_.updated as updated3_9_,
    endpoint0_.account_id as account_4_9_,
    endpoint0_.endpoint_sub_type as endpoint5_9_,
    endpoint0_.endpoint_type as endpoint6_9_,
    endpoint0_.description as descript7_9_,
    endpoint0_.enabled as enabled8_9_,
    endpoint0_.name as name9_9_,
    endpoint0_.status as status10_9_
from
    endpoints endpoint0_
where
    endpoint0_.endpoint_type=?
    and (
        endpoint0_.endpoint_sub_type in (
            ?
        )
    )
    and (
        endpoint0_.status not in  (
            ? , ?
        )
    ) for update
        of endpoint0_ skip locked
```